### DISCUSSION TITLE
Make Change Form Step "Specific step" value bindable...

### DISCUSSION THREAD

**Discussion by cfepdan** [Budibase/budibase#4967] [Ideas]
I'm building a multi-step form. Some of the steps are skippable based on certain conditions. It'd be helpful to have the number of the "Specific step" operation be a bindable (variable?) number (i.e., a field or state variable). This would be helpful for...

1) Naming the various steps in my form so it's easier to track where we're going, and
2) I'd like to have back button and a stack for tracking previous steps. This would allow users to click the back button to review their answers without encountering any of the steps we'd skipped.

**Comment by melohagan** [comment]
It may be a little tedious, but you could use <a href="https://docs.budibase.com/docs/app-state">App State</a> to store an index of what step you are on. 

So after every action to change form step, update your App State. This could be accessed wherever you need it. 

![Screenshot 2022-03-29 at 15 08 19](https://user-images.githubusercontent.com/101575380/160630291-de7d2f32-e4b7-47ab-996b-413aba8ecc9d.png)

**Comment by cfepdan** [comment]
Thank you @melohagan. This is indeed what I've done, except I implemented a stack to let me do backward/forward buttons, and I've decided to use show/hide conditions on containers with a state variable as an alternative to form steps. Somewhere along my development, something outside my direct control reset the form step. By using the persistent state variable, I get a little more control.

It works pretty well. Intuitively analogous to the back and forward buttons on a browser.

More details:
I have three state variables: form_step, step_stack_back, and step_stack_forward.

When the user presses the button to navigate to the next step, I push the current value of form_step to an array in step_stack_back, then I set form_step to the next step, and I clear step_stack_forward.

When the user presses the back arrow on my form, I push the current value of form_step to an array in step_stack_forward, pop the value from step_stack_back and save it to form_step.

When the user presses the forward arrow, I push the current value of form_step to step_stack_back, and pop the value from step_stack_forward and save it to form_step.

Right now I'm using some goofy javascript to manage the stack and convert back and forth between strings and arrays. I am pretty sure I can implement this with Handlebars functions though.

**Reply by NicolasFabre81** [reply]
I have quite the same problem with form steps. My issue is to be able to go directly to a specific step, having this step value stored in a variable.

But unfortunately there is no possible binding here :
<img width="575" alt="image" src="https://user-images.githubusercontent.com/28592069/188424321-c1167e5c-57d0-48c1-9079-2af74572de85.png">

**Reply by melohagan** [reply]
Ah that is a bummer! There is an issue open here: https://github.com/Budibase/budibase/issues/7063
